vsintegration/src/FSharp.Editor/LanguageService/FSharpCheckerExtensions.fs

```diff
@@ -60,19 +54,19 @@ type FSharpChecker with
                     
                 let! results =
                     match freshResults with
-                    | Ready x -> async.Return x
-                    | StillRunning worker ->
+                    | Some x -> async.Return (Some x)
+                    | None ->
                         async {
                             match allowStaleResults, checker.TryGetRecentCheckResultsForFile(filePath, options) with
                             | true, Some (parseResults, checkFileResults, _) ->
                                 return Some (parseResults, checkFileResults)
                             | _ ->
-                                return! worker
+                                return! parseAndCheckFile
```

Comment:
Please make `parseAndCheckFile` a function  `parseAndCheckFile ()`  to make things more obvious